instrumentation/jetty/jetty-9.1/src/main/java/io/opentelemetry/instrumentation/auto/jetty/Jetty91HandlerAdvice.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.instrumentation.auto.jetty;
+
+import static io.opentelemetry.instrumentation.auto.jetty.Jetty91HttpServerTracer.TRACER;
+
+import io.grpc.Context;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.instrumentation.auto.servlet.v3.AbstractCountingHttpServletResponse;
+import io.opentelemetry.instrumentation.auto.servlet.v3.CountingHttpServletRequest;
+import io.opentelemetry.instrumentation.auto.servlet.v3.TagSettingAsyncListener;
+import io.opentelemetry.instrumentation.auto.servlet.v3_1.Servlet31CountingHttpServletResponse;
+import io.opentelemetry.trace.Span;
+import java.lang.reflect.Method;
+import java.util.concurrent.atomic.AtomicBoolean;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import net.bytebuddy.asm.Advice;
+
+public class Jetty91HandlerAdvice {
```

Comment:
This is now identical to `Jetty8HandlerAdvice`, right? Can it be somehow deduplicated?

Reply:
It's almost identical -- the only differences are `TRACER` and the response wrapper type. Since it's an advice class everything must be static, no dependency injection, no passing parameters -- I don't really know how to parametrize it. Maybe someone here will have a good idea, I didn't, so I just copied the least code that I absolutely had to to make it work.

By the way, servlet 3.0 vs 3.1 has a very similar situation: the only difference is the response wrapper type.